### PR TITLE
[#187364719] GitHub actions with readme badge

### PR DESCRIPTION
**Pull request by Bananayosostene**
Integration GitHub actions with readme badge 

# Why is this important

This is required to automatically run tests and other preliminary build steps required to ship the next iteration of the product

# Setting up continuous integration

### CODE REVIEW

**Comment by munezerobagira** [comment]
can you check your title

**Comment by AngeloChristian1** [comment]
Add a pivotal id on the name of the PR

File: .github/workflows/node.js.yml [hunk 1]
```diff
@@ -0,0 +1,37 @@
+
+# This workflow will do a clean installation of node dependencies, cache/restore them, build the source code and run tests across different versions of node
+# For more information see: https://docs.github.com/en/actions/automating-builds-and-tests/building-and-testing-nodejs
+
+name: Node.js CI
+
+on:
+  push:
```
Comment (Yvan-David) [inline .github/workflows/node.js.yml]: it should be configured to develop branch or any branch so that it doesn't depend on your branch

**Review by Yvan-David — CHANGES_REQUESTED** [review]

**Review by Yvan-David — DISMISSED** [review]

**Review by EricNiyo73 — DISMISSED** [review]

File: src/math.js
Comment (munezerobagira) [inline src/math.js:1]: what is role of this js file?
It seem to be a build version of ts
can you remove it?

File: src/migrations/20240407175814-create-user.js
Comment (munezerobagira) [inline src/migrations/20240407175814-create-user.js:1]: **normally** deleting migration file is not recommended

File: test/math.test.js [hunk 2]
```diff
@@ -0,0 +1,18 @@
+"use strict";
+Object.defineProperty(exports, "__esModule", { value: true });
+const chai_1 = require("chai");
+const math_1 = require("../src/math");
+describe("Testing Index", () => {
+    it("should return 10 when have 8 + 2", () => {
```
Comment (munezerobagira) [inline test/math.test.js]: this also seem to be a build version of ts file,
can you remove it too?

**Review by munezerobagira — CHANGES_REQUESTED** [review]

**Review by munezerobagira — CHANGES_REQUESTED** [review]
can  you include this .nycrc.json configuration to make sure that the coverage it collect coverage from  all files in src
```json 
{
  "include": ["src/**/*.ts"],
  "exclude": ["node_modules/**", "coverage/**"],
  "all": true
}
```

File: src/dummy.ts [hunk 3]
```diff
@@ -1,15 +1,15 @@
...
+        return a-b;
+    }
+    multiply(a:number, b:number): number{
+        return a*b;
+    }
+    divide(a:number, b:number): number{
+        return a/b;
+    }
+}
+export default Calculator
```
Comment (munezerobagira) [inline src/dummy.ts]: we don't need this dummy you can delete it

File: src/math.ts
Comment (munezerobagira) [inline src/math.ts:1]: we can also remove this files

File: test/dummy.test.ts [hunk 4]
```diff
@@ -1,45 +1,45 @@
...
+    it("should return quotient", ()=>{
+        //arrange
+        const calc = new Calculator();
+
+        // act
+        const result = calc.divide(3,3);
+
+        //assert
+        expect(result).to.equal(1);
+    })
```
Comment (munezerobagira) [inline test/dummy.test.ts]:
```suggestion
```

File: test/math.test.ts [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+  });
+  it("Should return the 6 when we have 8 - 2", () => {
+    expect(sub(8, 2)).to.equal(6);
+  });
+  it("should return 16 when we have 8 * 2", () => {
+    expect(multiply(8, 2)).to.equal(16);
+  });
+  it("Should return 4 when we have 8 / 2", () => {
+    expect(div(8, 2)).to.equal(4);
+  });
```
Comment (munezerobagira) [inline test/math.test.ts]:
```suggestion

```
lets delete this math.test  its irrevant

**Review by munezerobagira — CHANGES_REQUESTED** [review]

File: src/dummy.ts (hunk 3 above)
Comment (munezerobagira) [reply src/dummy.ts]: we can use simple test as
expect(true).to.be(true)
all pass

File: test/math.test.ts (hunk 5 above)
Comment (Bananayosostene) [reply test/math.test.ts]: file was deleted

**Review by Yvan-David — APPROVED** [review]

File: src/math.ts
Comment (munezerobagira) [inline src/math.ts:1]: remove this file also

File: test/math.test.ts [hunk 6]
```diff
@@ -0,0 +1,20 @@
+import { expect } from "chai";
+import { sum, sub, multiply, div } from "../src/math";
+
+describe("Testing Index", () => {
+  it("should return 10 when have  8 + 2", () => {
+    expect(sum(8, 2)).to.equal(10);
```
Comment (munezerobagira) [inline test/math.test.ts]:
```suggestion
    expect(8+2).to.equal(10);
```
Can be as simple as that

File: test/math.test.ts [hunk 7]
```diff
@@ -0,0 +1,20 @@
...
+  });
+  it("Should return the 6  when we have 8 - 2", () => {
+    expect(sub(8, 2)).to.equal(6);
```
Comment (munezerobagira) [inline test/math.test.ts]:
```suggestion
    expect(8-2).to.equal(6);
```

File: test/math.test.ts [hunk 8]
```diff
@@ -0,0 +1,20 @@
...
+  });
+  it("should return 16 when we have 8 * 2", () => {
+    expect(multiply(8, 2)).to.equal(16);
+  });
+  it("Should return 4 when we have 8 / 2", () => {
+    expect(div(8, 2)).to.equal(4);
+  });
+});
```
Comment (munezerobagira) [inline test/math.test.ts]:
```suggestion
});
```

**Review by munezerobagira — CHANGES_REQUESTED** [review]

**Review by EricNiyo73 — APPROVED** [review]

**Review by munezerobagira — APPROVED** [review]

**Pull request merged**